besu/src/main/java/org/hyperledger/besu/controller/BesuControllerBuilder.java

```diff
@@ -345,7 +346,13 @@ public BesuController build() {
                   storageProvider.getStorageBySegmentIdentifier(
                       KeyValueSegmentIdentifier.CHAIN_PRUNER_STATE)),
               chainDataPruningBlocksRetained,
-              chainDataPruningFrequency);
+              chainDataPruningFrequency,
+              MonitoredExecutors.newBoundedThreadPool(
+                  ChainDataPruner.class.getSimpleName(),
+                  1,
+                  1,
+                  ChainDataPruner.MAX_PRUNING_WORKER,
```

Comment:
Isn't this queue size rather than number of worker threads?

Reply:
Updated.